### ISSUE TITLE
Only use read-only access on input files

### ISSUE THREAD

**Issue by VADemon** [DaMatrix/RegionMerger#1]
```
java.io.FileNotFoundException: X:\world\world\region\r.-1.-2.mca (Access is
 denied)
        at java.io.RandomAccessFile.open0(Native Method)
        at java.io.RandomAccessFile.open(RandomAccessFile.java:316)
        at java.io.RandomAccessFile.<init>(RandomAccessFile.java:243)
        at net.daporkchop.regionmerger.anvil.mojang.RegionFile.<init>(RegionFile.java:126)
        at net.daporkchop.regionmerger.World.getRegionOrNull(World.java:86)
        at net.daporkchop.regionmerger.RegionMerger.lambda$null$20(RegionMerger.java:508)
```
All region files are opened in rw unnecessarily. I wanted to set read-only on input files to be sure they don't get modified...
but
`file = new RandomAccessFile(path, "rw");`

**Comment by DaMatrix** [comment]
I'll fix this when I get home, I can get around this by simply fixing up my OverclockedRegionFile implementation (which uses memory mapping and NIO buffers for a heck of a lot better performance).

**Comment by DaMatrix** [comment]
A little update to this: I've now fixed up OverclockedRegionFile to allow read-only mapping, I think I'm going to rewrite this entire program to make full use of it (it's actually faster than the old one by a factor of 10 based on my tests).

**Comment by DaMatrix** [comment]
should be totally fixed as of release 0.1.8-SNAPSHOT.
